pika/adapters/select_connection.py

```diff
@@ -155,32 +157,34 @@ def _get_poller(self):
         return poller
 
 
-class SelectPoller(object):
-    """Default behavior is to use Select since it's the widest supported and has
-    all of the methods we need for child classes as well. One should only need
-    to override the update_handler and start methods for additional types.
+class _PollerBase(object):  # pylint: disable=R0902
+    """Base class for select-based IOLoop implementations"""
+    __metaclass__ = abc.ABCMeta
```

Comment:
@gmr, in order to make use of `abc.ABCMeta` in Python 3, we need to use the new Python 3 syntax that's not supported in Python 2. That new syntax (`MyClass(metaclass=abc.ABCMeta)`) prevents the module from loading when running in Python 2. To make it work without `six`, I had to wrap the Python 3 variant of the code in the `exec()` function so it won't break loads under Python 2, which isn't ideal (https://github.com/vitaly-krugl/pika/blob/897624da1fa6c9aeeb854aca3a9abf5dd4dcf148/pika/adapters/select_connection.py#L170). I am not sure how else to overcome this compatibility issue.